plugin/builtin/replacementtransformer/ReplacementTransformer.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2021 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+//go:generate pluginator
+package main
+
+import (
+	"sigs.k8s.io/kustomize/api/filters/replacement"
+	"sigs.k8s.io/kustomize/api/resmap"
+	"sigs.k8s.io/kustomize/api/types"
+	kyaml "sigs.k8s.io/kustomize/kyaml/yaml"
+	"sigs.k8s.io/yaml"
+)
+
+// Replace values in targets with values from a source
+type plugin struct {
+	Replacements []types.Replacement `json:"replacements,omitempty" yaml:"replacements,omitempty"`
+}
+
+//noinspection GoUnusedGlobalVariable
+var KustomizePlugin plugin
+
+func (p *plugin) Config(
+	_ *resmap.PluginHelpers, c []byte) (err error) {
+	p.Replacements = []types.Replacement{}
+	return yaml.Unmarshal(c, p)
+}
+
+func (p *plugin) Transform(m resmap.ResMap) (err error) {
+	var nodes []*kyaml.RNode
+	for _, r := range m.Resources() {
+		nodes = append(nodes, r.Node())
+	}
+	_, err = replacement.Filter {
```

Comment:
Discarding the results of the filter surprised me... I suppose this works because this particular filter implementation actually manipulates the RNodes, and we're passing in the RNodes that the resource map points to. I feel a bit uneasy about relying on that though, since the contract of the Filter interface is to give back the transformed results, not necessarily mutate the input. Is it difficult to replace the resource map's nodes with the "new" ones? If it is cumbersome, probably not worth changing given that the tests will fail if the filter stopped mutating the nodes. But if it is easy, strictly speaking that would seem more correct to me.

Reply:
I believe the results of filters are discarded in other transformers as well. I took a look at some of them and they use [Resource.ApplyFilter](https://github.com/kubernetes-sigs/kustomize/blob/0df531e7c6c5182f0d305bd1a43c760711a5c1dc/api/resource/resource.go#L570), which appears to use the Filter's return value only to check for node deletion. The results are never returned back to the transformer (for example, PatchStrategicMergeTarnsformer and ReplicaCountTransformer use this). 

Replacing the resource's map nodes with "new" ones would require substantial changes, most likely to resmap, reswrangler, and resource.  Reconstructing resources from modified nodes, and then reconstructing the resmap from the new resources would have some overhead and seems unnecessary if the filter modifies the nodes directly. 

I agree that using the results here is probably strictly more correct but I don't think it's worth it, especially considering that other transformers don't really use the results either.

Reply:
EDIT: Sorry, github is being strange and not sending my comment correctly. Hopefully this one works

Replacing the resource's map nodes with "new" ones would require substantial changes, most likely to resmap, reswrangler, and resource. Reconstructing resources from modified nodes, and then reconstructing the resmap from the new resources would have some overhead and seems unnecessary if the filter modifies the nodes directly.

I agree that using the results here is probably strictly more correct but I don't think it's worth it, especially considering that other transformers don't seem to use the results either.